ludwig/encoders/text_encoders.py

```diff
@@ -163,15 +205,21 @@ def __init__(
             pretrained_model_name_or_path
         )
         self.reduce_output = reduce_output
+        self.transformer.trainable = False
 
     def call(self, inputs, training=None, mask=None):
-        transformer_outputs = self.transformer(inputs, training=training)
+        transformer_outputs = self.transformer(
+            inputs, 
+            training=training,
+            attention_mask=mask,
+            token_type_ids=tf.zeros_like(inputs)
+        )
         if self.reduce_output == 'cls_pooled':
             hidden = transformer_outputs[1]
         else:
-            hidden = transformer_outputs[0]
+            hidden = transformer_outputs[0][:,1:-1,:]
```

Comment:
Some encoders are skipping the first and last element of the sequence with `[:,1:-1,:]`, some others don't. i guess it's an encoder specific thing right? You doublechecked the size of all of them?

Reply:
This is an encoder specific thing. Some of them append bos token or CLS token to the start of the tokenized input and append a eos or sep token to the end of the tokenized input. The indexing is to not consider the encoded output of those tokens.